test/katex-spec.js

```diff
@@ -1382,6 +1382,15 @@ describe("An HTML font tree-builder", function() {
         span = "<span class=\"mord mathbb\" style=\"color:blue;\">R</span>";
         expect(markup).toContain(span);
     });
+
+    it("should throw TypeError when the expression is of the wrong type", function() {
+        expect(function() {
+            katex.renderToString({badInputType: "yes"});
+        }).toThrowError(TypeError);
+        expect(function() {
+            katex.renderToString(["foo", {badInputType: "yes"}]);
+        }).toThrowError(TypeError);
+    });
```

Comment:
I'd say allow `String` objects but don't advertise that fact in any documentation, because it will be hardly relevant to most people. The way I see it, the distinction only matters up to the tokenization step. After that it's native strings either way. So if we have one single test case using `String` objects, I'd be confident that we actually support them just like we do native strings.